### PR TITLE
Query Editor: Fix field options not loading in

### PR DESCRIPTION
**Pull request by nmarrs**
Before 

https://github.com/user-attachments/assets/7bd008ad-8cc3-4d06-978b-60ccbd286632

After

https://github.com/user-attachments/assets/84a4fbc7-a3c0-4811-9ef4-e08a2d9beff2

Fixes https://github.com/grafana/athena-datasource/issues/291

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -7,6 +7,14 @@ export enum FormatOptions {
   Logs,
 }
 
+export enum QueryEditorFieldType {
```
Comment (nmarrs) [inline src/types.ts:10]: This is an unrelated clean up / refactor to avoid magic strings 🧙

File: src/QueryEditorForm.tsx [hunk 2]
```diff
@@ -45,40 +60,49 @@ export function QueryEditorForm(props: Props) {
...
 
-  const fetchRegions = () =>
-    props.datasource.getRegions().then((regions) => appendTemplateVariables(templateVariables, regions));
-  const fetchCatalogs = () =>
-    props.datasource
+  const fetchRegions = async () =>
+    await props.datasource
+      .getRegions()
+      .then((regions) => appendTemplateVariables(templateVariables, regions))
+      .finally(() => setNeedsUpdate(true));
```
Comment (nmarrs) [inline src/QueryEditorForm.tsx:67]: Updated to be async / await functions to match redshift, but doesn't impact functionality / fix at all

File: src/QueryEditorForm.tsx [hunk 3]
```diff
@@ -36,6 +40,17 @@ export function QueryEditorForm(props: Props) {
...
+  /*
+    State used force the component to re-render when fetching
+    editor field options so they show up in the select component
+  */
+  const [needsUpdate, setNeedsUpdate] = useState(false);
+  useEffect(() => {
+    if (needsUpdate) {
+      setNeedsUpdate(false);
+    }
+  }, [needsUpdate]);
```
Comment (nmarrs) [inline src/QueryEditorForm.tsx:52]: This state is what fixes the issue, allowing us to force re-render the form component after loading in field options

**Review by idastambuk — APPROVED** [review]
🥳
